File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Paste text here: ")
sent = 0
words = 0
letters = 0

for i in range(len(text)):
    if text[i] == ' ':
        words += 1
    if text[i] == '.' or text[i] == '!' or text[i] == '?':
        sent += 1
    if (text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z'):
        letters += 1
words += 1

L = letters / words * 100
S = sent / words * 100

index = round(0.0588 * L - 0.296 * S - 15.8)

if index > 15:
    print('Grade 16+')
elif index < 2:
    print('Before Grade 1')
else:
    print('Grade: ', index)

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{

    int ind;
    float L, S, index, letters = 0, spaces = 0, words = 0, sent = 0;

    string text = get_string("Paste text here: \n");

    for (int i = 0, n = strlen(text); i < n; i++)
    {

        if (isalpha(text[i]))
        {
            letters++;
        }
        if (text[i] == ' ')
        {
            spaces++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sent++;
        }

    }
    words = spaces + 1;
    
    
    L = letters / words * 100;
    S = sent / words * 100;
    
    
    index = 0.0588 * L - 0.296 * S - 15.8;
    ind = round(index);
    if (ind > 15)
    {
        printf("Grade 16+\n");
    }
    else if (ind < 2)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", ind);
    }
}